File: Remove Duplicates from Sorted Array II/Remove Duplicates from Sorted Array II.cpp
class Solution {
public:
    int removeDuplicates(int A[], int n) {
        if(!n)      return 0;
        
        int prev = A[0];
        int ptr = 1;
        int twiceFlag = false;
        int len = n;
        for(unsigned int i = 1; i < n; ++i){
            if(prev == A[i]){
                if(twiceFlag){
                    // ptr ain't moving
                    --len;
                }
                else{
                    twiceFlag = true;
                    A[ptr] = A[i];
                    ++ptr;
                    prev = A[i];
                }
            }
            else{
                twiceFlag = false;
                A[ptr] = A[i];
                ++ptr;
                prev = A[i];
            }
        }
        
        return len;
    }
};

File: Remove Duplicates from Sorted Array II/Remove Duplicates from Sorted Array II.java
public class Solution {
    public int removeDuplicates(int[] A) {

        if(A.length <= 1)       return A.length;

        int prev = A[0];
        int currIndex = 0;
        boolean twice = false;
        
        for(int i = 1; i < A.length; ++i){
            if(A[currIndex] == A[i]){
                if(twice){
                    
                }
                else{
                    twice = true;
                    ++currIndex;
                    A[currIndex] = A[i];
                }
            }
            else{
                twice = false;
                ++currIndex;
                A[currIndex] = A[i];
            }
        }
        
        return currIndex + 1;
    }
}
